### PR TITLE
ENG-580 Roles form UX specs + current user auth reducer change

### PR DESCRIPTION
**Pull request by jeffgo10**
includes:
 - roles form permission auto-enable switch when all functions ("superuser") is selected
 - reducer change on adding current user roles
 - fetching current user authority relocated with fetch permission list now in ApiManager

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ichalagashvili**

If they are not already assigned, you can assign the PR to them by writing `/assign @ichalagashvili` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/state/current-user-auth/reducer.js [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
   switch (action.type) {
-    case SET_CURRENT_USER_AUTH:
-      return [...action.payload];
+    case SET_CURRENT_USER_AUTH: {
+      const { result, allPermissions } = action.payload;
+      const authMaps = result.map(auths => (
+        stringToArr(auths.permissions ? auths.permissions : get(auths, 'role', []))
+      ));
+      const roles = Array.from(new Set(flatten(authMaps)));
+      return { ...state, auth: result, roles: roles.includes('superuser') ? allPermissions : roles };
```
Comment (gidesan) [inline src/state/current-user-auth/reducer.js:21]: Please refactor all the `superuser` occurrences to a constant.

File: src/app-init/apiManager.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { ApiProvider } from '@entando/apimanager';
+import { fetchPermissions } from 'state/permissions/actions';
+import { fetchUserAuth, clearCurrentUserAuth } from 'state/current-user-auth/actions';
```
Comment (gidesan) [inline src/app-init/apiManager.js:5]: Why `UserAuth` if it's about group permission? This name is misleading

File: src/state/current-user-auth/reducer.js [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
+import { get, flatten } from 'lodash';
 
-const reducer = (state = [], action = {}) => {
+const stringToArr = val => (
+  Array.isArray(val) ? val : [val]
+);
+
+const initialState = {
+  auth: [],
+  roles: [],
```
[the remark this answers is not in the document]
Comment (jeffgo10) [reply src/state/current-user-auth/reducer.js]: I have just removed `auth` sub state and maintain "roles" with new sub state name `loggedUser` but now under in existing `permissions` state

File: src/state/current-user-auth/reducer.js (hunk 1 above)
Comment (jeffgo10) [reply src/state/current-user-auth/reducer.js]: constant now under `state/permissions/const.js`

File: src/app-init/apiManager.js (hunk 2 above)
Comment (jeffgo10) [reply src/app-init/apiManager.js]: meant for "UserAuthorities" but I made some changes by moving all them on `permission` state

**Comment by jeffgo10** [comment]
@gidesan naming has now been changed but most significant change here is that `currentUserAuth` has been retired and moved to existing `permissions` state. I thought that both of basic permission listing and current user permissions should be under one wing

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/710#pullrequestreview-417534619" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**